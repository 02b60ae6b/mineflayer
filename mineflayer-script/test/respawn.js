var diedYet = false;

mf.setTimeout(function() {
    mf.print("Test timed out.\n");
    mf.exit();
}, 8000);

mf.onDeath(function() {
    diedYet = true;
    mf.print("Dead, respawning.\n");
    mf.respawn();
});
mf.onConnected(function() {
    mf.print("Connected.\n");
});
mf.onSpawn(function() {
    if (diedYet) {
        mf.print("I have respawned.\n");
        mf.exit();
    } else {
        mf.setTimeout(function() {
            mf.print("I have spawned, killing myself.\n");
            mf.chat("/kill");
        }, 4000);
    }
});
