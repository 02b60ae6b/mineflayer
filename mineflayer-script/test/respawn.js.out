Connected.
I have spawned, killing myself.
Dead, respawning.
I have respawned.
